lightning/src/chain/package.rs

```diff
@@ -641,6 +661,31 @@ impl PackageTemplate {
 		let output_weight = (8 + 1 + destination_script.len()) * WITNESS_SCALE_FACTOR;
 		inputs_weight + witnesses_weight + transaction_weight + output_weight
 	}
+	#[cfg(anchors)]
+	pub(crate) fn construct_malleable_package_with_external_funding<Signer: Sign>(
+		&self, onchain_handler: &mut OnchainTxHandler<Signer>,
+	) -> (Transaction, Vec<ExternalHTLCClaim>) {
+		debug_assert!(self.requires_external_funding());
+		let mut aggregate_tx = Transaction {
```

Comment:
If we got correctly the aggregation flags in `channelmonitor` we should only ever aggregate HTLC-preimage transactions, never HTLC-timeout together (as otherwise it's unsafe with pinning). I think the flag is correct L2634 for building package against counterparty commitment, however I wonder if we got it correctly L2693 for building package against holder commitment. At least post-anchor, we can and should aggregate our HTLC-Success transactions together ?